File: java apni kaksha coding sol/constellation.java
import java.util.*;
public class constellation {
    public static Boolean check(String[][] mat1,String[][]mat2){
        for(int row = 0;row<3;row++){
            for(int col=0;col<3;col++){
                if(!mat1[row][col].equals(mat2[row][col])){
                    return false;
                }
            }
        }
        return true;
    }

    public static String find_charr(String [][]image){
        String [][]A = {{".","*","."},{"*","*","*"},{"*",".","*"}};
        String [][]E = {{"*","*","*"},{"*","*","*"},{"*","*","*"}};
        String [][]I = {{"*","*","*"},{".","*","."},{"*","*","*"}};
        String [][]O = {{"*","*","*"},{"*",".","*"},{"*","*","*"}};
        String [][]U = {{"*",".","*"},{"*",".","*"},{"*","*","*"}};
         
        if(check(image,A))
            return "A";

        else if(check(image,E))
            return "E";

        else if(check(image,I))
            return "I";

        else if(check(image,O))
            return "O";

        else if(check(image,U))
            return "U";
        else{
            return "";
        }
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        String grid[][] = new String[3][N];
        String image[][] = new String[3][3];
        for(int row = 0;row < 3; row++){
            int col=0;
            while(col<N){
            String s = sc.next();
                if(s != " "){
                    grid[row][col] = s;
                    col++;
                }
            }
        }

        
        for(int col=0 ;col<N; col++){
            if(grid[0][col].equals("#")){
                System.out.print("#");
            }
            else if(grid[0][col].equals(".") && grid[1][col].equals(".") && grid[2][col].equals(".")){
                continue;
            }
            else{
               
                for(int cur_row = 0; cur_row<3 ; cur_row++){
                    for(int cur_col = 0;cur_col<3;cur_col++){
                        image[cur_row][cur_col] = grid[cur_row][cur_col + col];
                    }
                }
                 
                String get_char = find_charr(image);
                System.out.print(get_char);
                col+=2;
            }
            
        }
        
    }
}

File: code/constellation.cpp
#include<bits/stdc++.h>
using namespace std;
char A[][3] = {{'.','*','.'},{'*','*','*'},{'*','.','*'}};
char E[][3] = {{'*','*','*'},{'*','*','*'},{'*','*','*'}};
char I[][3] = {{'*','*','*'},{'.','*','.'},{'*','*','*'}};
char O[][3] = {{'*','*','*'},{'*','.','*'},{'*','*','*'}};
char U[][3] = {{'*','.','*'},{'*','.','*'},{'*','*','*'}};

char check(char image[][3]){
	int flag = 1;
	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(A[i][j] != image[i][j]){
				flag = 0;
				break;
			}
		}
		if(flag == 0){
			break;
		}
	}
	if(flag == 1){
		return 'A';
	}


	flag = 1;
	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(E[i][j] != image[i][j]){
				flag = 0;
				break;
			}
		}
		if(flag == 0){
			break;
		}
	}
	if(flag == 1){
		return 'E';
	}

	flag = 1;
	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(I[i][j] != image[i][j]){
				flag = 0;
				break;
			}
		}
		if(flag == 0){
			break;
		}
	}
	if(flag == 1){
		return 'I';
	}

	flag = 1;
	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(O[i][j] != image[i][j]){
				flag = 0;
				break;
			}
		}
		if(flag == 0){
			break;
		}
	}
	if(flag == 1){
		return 'O';
	}

	flag = 1;
	for(int i=0;i<3;i++){
		for(int j=0;j<3;j++){
			if(U[i][j] != image[i][j]){
				flag = 0;
				break;
			}
		}
		if(flag == 0){
			break;
		}
	}
	if(flag == 1){
		return 'U';
	}

	
}
int main(){
	int n;
	cin>>n;
	char a[3][n];
	char image[3][3];
	for(int i=0;i<3;i++){
		for(int j=0;j<n;j++){
			cin>>a[i][j];
		}
	}

	for(int i=0;i<n;i++){
		if(a[0][i] == '#'){
			cout<<'#';
		}else if(a[0][i]=='.' && a[1][i] == '.' && a[2][i] =='.'){
			continue;
		}else{
			int col = i;
			for(int j=0;j<3;j++){
				for(int k=i;k<i+3;k++){
					image[j][k-col] = a[j][k];
				}
			}
			
			cout<<check(image);
			i+=2;
		}
	}
	return 0;
}
